src/main/java/com/study_buddy/study_buddy/controller/LoginController.java

```diff
@@ -50,24 +61,85 @@ public Map<String, String> register(@RequestBody Register data) {
         String lastName = data.getLastName();
         String hashedPassword = data.getHashedPassword();
         StudyRole studyRole = data.getStudyRole();
-        Map<String, String> returnData = Map.of("firstName", firstName, "lastName", lastName, "email", email,
-                "hashedPassword", hashedPassword.toString(), "studyRole", studyRole.toString());
-
-        // TODO: Save this user to the database
-       /* User user = new User(email, hashedPassword, firstName, lastName, studyRole);
+        System.out.println(hashedPassword.toString());
+        // Example logic for generating a JWT after registering a user
+        JwtService jwtService = new JwtService();
+        String token = jwtService.generateToken(data.getEmail());  // Implement JWT generation
 
+        // Save this user to the database
+        User user = new User(token, LocalDateTime.now(), email, firstName, lastName, "", "", hashedPassword, "", studyRole, LocalDateTime.now());
         userRepository.createUser(user);
-*/
-        return returnData;
+
+        // Create Authentication object
+        List<SimpleGrantedAuthority> authorities = List.of(new SimpleGrantedAuthority("ROLE_USER"));  // You can adjust the authorities as needed
+        UsernamePasswordAuthenticationToken authentication = new UsernamePasswordAuthenticationToken(
+                user.getEmail(), null, authorities);
+
+        // Set the authentication in the SecurityContext
+        SecurityContext context = SecurityContextHolder.createEmptyContext();
+        context.setAuthentication(authentication);
+        SecurityContextHolder.setContext(context);
+
+        Map<String, String> returnData = Map.of(
+                "firstName", firstName,
+                "lastName", lastName,
+                "email", email,
+                "hashedPassword", hashedPassword.toString(),
+                "studyRole", studyRole.toString(),
+                "token", token,
+                "message", "Registration successful"
+        );
+
+        // Send the token back to the client
+        return ResponseEntity.ok()
+                .header(HttpHeaders.AUTHORIZATION, "Bearer " + token)  // Optional: set in header
```

Comment:
remove comment as it is misleading, returning the token is not optional :) (and it's set in the header already)